pkg/operator/k8sutil/deployment.go

```diff
@@ -254,14 +255,31 @@ func addLabel(key, value string, labels map[string]string) {
 	labels[key] = value
 }
 
-func CreateDeployment(name, namespace string, clientset kubernetes.Interface, dep *apps.Deployment) error {
-	_, err := clientset.AppsV1().Deployments(namespace).Create(dep)
+func CreateDeployment(clientset kubernetes.Interface, name, namespace string, modifiedDeployment *apps.Deployment) error {
+	currentDeployment, err := clientset.AppsV1().Deployments(namespace).Create(modifiedDeployment)
 	if err != nil {
 		if k8serrors.IsAlreadyExists(err) {
-			_, err = clientset.AppsV1().Deployments(namespace).Update(dep)
-		}
-		if err != nil {
-			return fmt.Errorf("failed to start %s deployment: %+v\n%+v", name, err, dep)
+			// Check whether the current deployement and newly generated one are identical
+			patchResult, err := patch.DefaultPatchMaker.Calculate(currentDeployment, modifiedDeployment)
```

Comment:
Let's make this a new method rather than affecting all callers of the k8sutil.CreateDeployment() method. Only the CSI pods need to worry about this. Other places that need to worry about comparing for the spec diff are already doing it. But by default everything else should rely on just updating the deployment and let K8s update if needed.

Reply:
Currently csi is the only consumer of this function, how about passing a extra argument?  If required I can create a new function for this one, but this function will be left unused

Reply:
Maybe just renaming it to CreateOrUpdateDeployment()

Reply:
Instead of implementing the deletion and creation, looks like the [Recreate](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#recreate-deployment) update strategy on the setting may work for us. Then the pods from the previous spec will be updated before the new ones are created.

Reply:
Thanks for the pointer, this works as expected